### DISCUSSION TITLE
An values format error occurred while using Traditional Chinese

### DISCUSSION THREAD

**Discussion by eric2chek** [dbarzin/deming#113] [Q&A] [answered]
Need help~~
- An error occurred when I tried to add a attribute with Chinese 
- After testing the following format combinations, none of them can be written to the database.
  - #機密#可用性#完整性
  - #機密 #可用性 #完整性
  
- Mysql config follow the guidelines
  - CREATE DATABASE deming CHARACTER SET utf8 COLLATE utf8_general_ci;

![2024-07-31_051616](https://github.com/user-attachments/assets/292c3832-8e56-4f9e-bb9e-d67b0b0cdb91)

**Comment by eric2chek** [comment]
the charset is utf8mb4_unicode_ci

![2024-07-31_054609](https://github.com/user-attachments/assets/2818f511-5078-40e0-bf73-d9f2a247338d)

**Comment by eric2chek** [comment, marked as answer]
found a solution

- sudo nano /var/www/deming/app/Http/Controllers/AttributeController.php

> -  /* remove =>     |regex:/^(#[a-zA-Z0-9_èéÉ\']+ *)*$/
'values' => 'required|regex:/^(#[a-zA-Z0-9_èéÉ\']+ *)*$/|max:4000',  
*/
> -  'values' => 'required|max:4000',

- both function about  store 、 update
- it working now
 
![2024-08-03_162032](https://github.com/user-attachments/assets/b45d7e8c-1dee-4c51-bce0-63353317897a)

**Comment by dbarzin** [comment]
Thanks a lot for this. I will change the code with your proposal.

**Comment by dbarzin** [comment]
So after some tests, I come with another solution that works better : 

    'values' => "required|regex:/^(#[\p{L}\p{M}\p{N}'_-]+ *)*$/u|max:4000",

It has been tested with

     #Bonjour #Éducation #DéjàVu #你好
     #Café #Mélange #Résumé #开发
     #Tag123 #Hello_World #L'été

**Reply by eric2chek** [reply]
thanks for your work !!!

**Comment by dbarzin** [comment]
Commited here : https://github.com/dbarzin/deming/pull/133
